router.go

```diff
@@ -313,6 +314,92 @@ func (*App) copyRoute(route *Route) *Route {
 	}
 }
 
+func (app *App) normalizePath(path string) string {
+	if path == "" {
+		path = "/"
+	}
+	if path[0] != '/' {
+		path = "/" + path
+	}
+	if !app.config.CaseSensitive {
+		path = utils.ToLower(path)
+	}
+	if !app.config.StrictRouting && len(path) > 1 {
+		path = utils.TrimRight(path, '/')
+	}
+	return RemoveEscapeChar(path)
+}
+
+// RemoveRoute is used to remove a route from the stack by path.
+// This only needs to be called to remove a route, route registration prevents duplicate routes.
+// You should call RebuildTree after using this to ensure consistency of the tree.
+func (app *App) RemoveRoute(path string, methods ...string) {
+	// Normalize same as register uses
+	norm := app.normalizePath(path)
+
+	pathMatchFunc := func(r *Route) bool {
+		return r.path == norm // compare private normalized path
+	}
+	app.deleteRoute(methods, pathMatchFunc)
+}
+
+// RemoveRouteByName is used to remove a route from the stack by name.
+// This only needs to be called to remove a route, route registration prevents duplicate routes.
+// You should call RebuildTree after using this to ensure consistency of the tree.
+func (app *App) RemoveRouteByName(name string, methods ...string) {
+	matchFunc := func(r *Route) bool { return r.Name == name }
+	app.deleteRoute(methods, matchFunc)
+}
+
+// RemoveRouteFunc is used to remove a route from the stack by a custom match function.
+// This only needs to be called to remove a route, route registration prevents duplicate routes.
+// You should call RebuildTree after using this to ensure consistency of the tree.
+// Note: The route.Path is original path, not the normalized path.
+func (app *App) RemoveRouteFunc(matchFunc func(r *Route) bool, methods ...string) {
+	app.deleteRoute(methods, matchFunc)
+}
+
+func (app *App) deleteRoute(methods []string, matchFunc func(r *Route) bool) {
+	if len(methods) == 0 {
```

Comment:
since the methods are required for all parent methods
we should not make it optional

maybe something like 
```go
func (app *App) RemoveRouteFunc(matchFunc func(r *Route) bool, method string, methods ...string)
```

or 

```go
func (app *App) RemoveRouteFunc(matchFunc func(r *Route) bool, method string[])
```
is better

Reply:
Or we use all registered Methods as default -> then its okay to have it optional

Reply:
Yes i will use default methods when methods is empty